### ISSUE TITLE
Precompress static files

### ISSUE THREAD

**Issue by dmos62** [centerofci/mathesar#2411]
Branched off of #2204. #2204 is about compressing on the fly and this is about providing precompressed versions of static files so that 1) Caddy doesn't have to encode each response and 2) so that we can use Brotli.

For context, PR implementing #2204: https://github.com/centerofci/mathesar/pull/2410.

**Comment by silentninja** [comment]
@dmos62 I don't think we need to fix this issue before our launch. Performance Improvement gained by Compressing on the fly should be more than enough for our public release. So can we move this to Backlog?

**Comment by dmos62** [comment]
@silentninja I don't know why #2204 was considered a launch-blocker and so can't judge whether this is (this is split off of #2204). @pavish @kgodey should we deprioritize this?

**Comment by pavish** [comment]
@dmos62 Only the compression on the fly was a launch blocker. Brotli compression (or any precompression) was part of the same original ticket, but lower priority. I can we can deprioritize this since we've split them up.

**Comment by dmos62** [comment]
Discussed this with @silentninja briefly. Marking this as a good first issue.

The precompression should be triggered arund the `collectstatic` call in the `Dockerfile`. We'd normally want it in a script that's not Docker-specific, but we don't yet have that, so calling it from the `Dockerfile` will do for the moment.

**Comment by thesujai** [comment]
@dmos62 I dont see a `collectstatic` call in `Dockerfile`

**Comment by dmos62** [comment]
@thesujai thanks for the feedback. I only see [a `collectstatic` call in `install.py`](https://github.com/centerofci/mathesar/blob/6de148a7d582bb5ec0efb6664f10fe916bced467/install.py#L27). Is this the call you meant @silentninja?

**Comment by SafeEHA** [comment]
Hello @dmos62 i see this issue hasn't been resolved, I would like to work on it, please

**Comment by dmos62** [comment]
@SafeEHA go for it. We're not assigning issues to new contributors atm, so as to not reduce the pool of issues available to GSoC candidates.

**Comment by rai-ankur** [comment]
Hello @dmos62 i see this issue hasn't been resolved, could you please assign this issue to me ?

**Comment by seancolsen** [comment]
@rai-ankur You're welcome to work on this and submit a PR, but in general we don't assign tickets to new contributors, as noted in our [Contributor Guide](https://github.com/centerofci/mathesar/blob/develop/CONTRIBUTING.md). Once we've already merged a PR of yours, then we'll be happy to assign tickets to you after that. The reason for this policy is that in the past it has been a maintenance burden for us to manage ticket assignments while having an overwhelming volume of people fail to submit their first PR when initially expressing intent to contribute.

**Comment by mathemancer** [comment]
@rai-ankur Please let me know if you get stuck.

**Comment by cmagapu** [comment]
Hi @dmos62 please do assign this to me if it hasn't been resolved. I'm new to open source and would like to contribute

**Comment by seancolsen** [comment]
@cmagapu We don't assign PRs to new contributors right now but you're welcome to submit a PR for this.
